Updates so that neurodemo runs with latest PyQt and pyqtgraph

JhouLab opened the pull request.

pbmanis commented:
Most of these changes align the code with the current usage in pyqtgraph, and look good to me.
There is one however, the I think Luke should comment on, which is calling the "protected" _copy_attrs function in pyqtgraph.Qt. Whether it should be called in this way, or perhaps just the function included separately here (which would make the intent clearer) should be considered.

JhouLab commented:
Thanks Paul, glad the changes look OK. Would be glad to hear Luke's thoughts on _copy_attrs. It's clearly a hack on my part, so would love to see a more elegant solution!

JhouLab commented:
I added a .spec file for PyInstaller to build a Windows .EXE, and fixed an issue that would cause crash if user adds a new ion channel after stopping a run. I hope these make sense. My understanding of neurodemo is still rudimentary, so please let me know if I've missed something. Overall, this is a fantastic tool, kudos to all who made it work!

campagnola commented:
Many thanks @JhouLab! I took a more drastic approach (relative to _copy_attrs) and added a neurodemo.qt wrapper module; everything else in the package will now get Qt names from there. This should make it easier to manage Qt changes in the future.

I also fixed an issue with the channels being drawn all black and replaced the splash image (which contained evidence of that bug). 

Also also, fixed the analyzer window so it appears to be working again as well (it needed a `values`=>`limits` change, similar to the ones you fixed elsewhere).

JhouLab commented:
Your updates look great. Glad to have been of some small help!

pbmanis commented:
@Jhoulab - thanks from me as well. I have been using this as an aid when teaching cellular electrophysiology for the past 3 years, and Luke used it previously when he was a TA in the course. Getting the few latent bugs out (and likely some I introduced when resurrecting the program a couple of years ago) is a good thing! I had not updated from the recent changes in pyqtgraph, some of which - and especially the value/limits change -  have percolated throughout my other programs.
I also have a document that goes with this program that is a guide to its use as well as some exercises for users to go through - similar to the old "Neurons in Action" program.  Although it is not complete, I'll put that up at some point in the near future. 
For the class, I distributed win x86_64 (.exe) programs, as well as MacOX X86_64 and M1 (.app) versions. If someone was running Linux (and no one was), I suspect they would just run from a local environment. I think all the builder and .spec files are still in the repository, although I haven't tried to rebuild them in the past 2 years. Having the pre-built versions is essential...

JhouLab commented:
This is indeed a great teaching tool that is not just informative but also (gasp) fun to play with. Love the ultra-smooth animations, and I'm glad Luke mentioned it in his email.

I plan to use it next month when I'm teaching (for the first time) a couple of lectures in slice ephys to our PhD students. If you have a guide document, and can upload it before Feb 7, that would be great.

I do have one question that perhaps the guide document will cover: how exactly does arrow size relate to the current? Is it arrow-length = log(current)? Or something else?

- Tom Jhou
